Disable logging to $HOME/.sdformat/sdformat.log

Once https://github.com/osrf/sdformat/issues/334 is resolved, we can
probably remove this patch from Drake.

This patch was created by `diff -u` on a v9.2.0 checkout of sdformat
after editing the source file locally.

--- a/src/Console.cc	2020-04-02 14:55:20.000000000 -0400
+++ b/src/Console.cc	2020-08-10 10:39:40.290726763 -0400
@@ -45,6 +45,7 @@
 Console::Console()
   : dataPtr(new ConsolePrivate)
 {
+#if 0
   // Set up the file that we'll log to.
 #ifndef _WIN32
   const char *home = std::getenv("HOME");
@@ -72,6 +73,7 @@
   }
   std::string logFile = sdf::filesystem::append(logDir, "sdformat.log");
   this->dataPtr->logFileStream.open(logFile.c_str(), std::ios::out);
+#endif
 }
 
 //////////////////////////////////////////////////
